### PR TITLE
Unmarshall numbers in options to number not float64

### LINKED ISSUE #302: integer values should be passed as is
when we specify options(collector, query, agent) in CR files when creating actual service, options(integer values) are passed in `scientific notation format`, specific jaeger service throws an exception,

CR file:
```yaml
apiVersion: jaegertracing.io/v1
kind: Jaeger
metadata:
  name: jaegerqe
spec:
  ingress:
    security: none
  strategy: production
  collector:
    replicas: 1
    image: jaegertracing/jaeger-collector:latest
    options:
      metrics-backend: expvar
      collector:
        num-workers: 50
        queue-size: 2000
      es:
        bulk:
          size: 5000000
          workers: 1
          flush-interval: 200ms
  query:
    replicas: 1
    image: jaegertracing/jaeger-query:latest
    options:
      metrics-backend: expvar
      query:
        port: 16686
  agent:
    strategy: sidecar
    image: jaegertracing/jaeger-agent:latest
    options:
      metrics-backend: expvar
  storage:
    type: elasticsearch
    esIndexCleaner:
      enabled: false
    sparkDependencies:
      enabled: false
    options:
      es:
        server-urls: http://elasticsearch:9200
```

Exception in collector:
```
Error: invalid argument "5e+06" for "--es.bulk.size" flag: strconv.ParseInt: parsing "5e+06": invalid syntax
invalid argument "5e+06" for "--es.bulk.size" flag: strconv.ParseInt: parsing "5e+06": invalid syntax
```

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves #302 

Signed-off-by: Pavol Loffay <ploffay@redhat.com>

### CODE REVIEW

**Comment by jpkrohling** [comment]
This change is [Reviewable](https://reviewable.io/reviews/jaegertracing/jaeger-operator/308)

File: pkg/apis/jaegertracing/v1/options.go [hunk 1]
```diff
@@ -38,7 +39,11 @@ func (o *Options) Filter(prefix string) Options {
 // UnmarshalJSON implements an alternative parser for this field
 func (o *Options) UnmarshalJSON(b []byte) error {
 	var entries map[string]interface{}
-	json.Unmarshal(b, &entries)
+	d := json.NewDecoder(bytes.NewReader(b))
+	d.UseNumber()
```
Comment (pavolloffay) [inline pkg/apis/jaegertracing/v1/options.go]: A bit of context JSON spec defines numbers as `float64` so all number are by default unmarshalled to `float64`. Our `Options.Args()` methods uses `fmt.Sprintf(%d, val)` to convert the value to string - the `%v` formats the float into exponent notation e.g. `5e+06`

Comment (objectiser) [reply pkg/apis/jaegertracing/v1/options.go]: Thanks for the details - would it be possible to provide a test?

Comment (objectiser) [reply pkg/apis/jaegertracing/v1/options.go]: Including one that fails the decode, just to improve the code coverage.

Comment (pavolloffay) [reply pkg/apis/jaegertracing/v1/options.go]: Apart from it is it ok?

Comment (objectiser) [reply pkg/apis/jaegertracing/v1/options.go]: Yes

**Pull request merged**